platform/cosmos/stake.go

```diff
@@ -38,15 +42,33 @@ func (p *Platform) GetValidators() (blockatlas.ValidatorPage, error) {
 }
 
 func (p *Platform) GetDetails() blockatlas.StakingDetails {
-	//TODO: Find a way to have a dynamic
 	return blockatlas.StakingDetails{
-		Reward:        blockatlas.StakingReward{Annual: 11},
+		Reward: blockatlas.StakingReward{
+			Annual: p.GetMaxAPR(),
+		},
 		MinimumAmount: blockatlas.Amount("0"),
 		LockTime:      1814400,
 		Type:          blockatlas.DelegationTypeDelegate,
 	}
 }
 
+func (p *Platform) GetMaxAPR() float64 {
+	validators, err := p.GetValidators()
```

Comment:
I think this API is going to be slow. Can we try to cache something? like the `GePool` or `GetInflation`request

Reply:
Take a look at my comment in the PR